### PR TITLE
Add GN Build

### PR DESCRIPTION
**Pull request by dj2**
This CL adds the necessary infrastructure to create a GN build. Only the libraries are generated at this point. Fuzzer targets have been added to show that building and linking of the static_libraries works correctly.

### CODE REVIEW

File: testing/test.gni [hunk 1]
```diff
@@ -0,0 +1,413 @@
+# Copyright 2018 The Chromium Authors. All rights reserved.
```
Comment (dj2) [inline testing/test.gni]: This file is copied from Chromium, so I left the Chromium copyright and added the license text below.

Comment (dneto0) [reply testing/test.gni]: Thanks for calling that out.  Seems ok to me.

File: testing/fuzzers/spvtools_opt_fuzzer.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  spvtools::Optimizer optimizer(SPV_ENV_UNIVERSAL_1_3);
+  optimizer.SetMessageConsumer([](spv_message_level_t, const char*,
+                                  const spv_position_t&, const char*) {});
+
+  std::vector<uint32_t> input;
+  input.resize(size >> 2);
+
+  size_t count = 0;
```
Comment (dneto0) [inline testing/fuzzers/spvtools_opt_fuzzer.cpp]: I would have gone with a std::memcpy.   Meh.

File: testing/fuzzers/spvtools_opt_fuzzer.cpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdint>
+
+#include "spirv-tools/optimizer.hpp"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  spvtools::Optimizer optimizer(SPV_ENV_UNIVERSAL_1_3);
```
Comment (dneto0) [inline testing/fuzzers/spvtools_opt_fuzzer.cpp]: The optimizer won't do anything unless you register some passes.  So let's expand this later.

**Comment by dneto0** [comment]
Rebased, squashed, and pushed into master as 3b9f197be1

**Comment by dneto0** [comment]
Yep, I rebased and squashed on my machine but failed to push ...  

I've rebased again and pushed as a114e1f30c34

**Comment by Kangz** [comment]
Thank you!

**Pull request closed without merging**